addons/website_sale/controllers/delivery.py

```diff
@@ -187,6 +187,55 @@ def _get_rate(carrier, order, is_express_checkout_flow=False):
                     rate['price'] = taxes['total_included']
         return rate
 
+    def order_lines_2_google_api(self, order_lines):
+        """ Transforms a list of order lines into a dict for google analytics """
+        order_lines_not_delivery = order_lines.filtered(lambda line: not line.is_delivery)
+        return super(WebsiteSaleDelivery, self).order_lines_2_google_api(order_lines_not_delivery)
+
+    def order_2_return_dict(self, order):
+        """ Returns the tracking_cart dict of the order for Google analytics """
+        ret = super(WebsiteSaleDelivery, self).order_2_return_dict(order)
+        delivery_line = order.order_line.filtered('is_delivery')
+        if delivery_line:
+            ret['shipping'] = delivery_line.price_unit
+        return ret
+
+    def _get_express_shop_payment_values(self, order, **kwargs):
+        values = super(WebsiteSaleDelivery, self)._get_express_shop_payment_values(order, **kwargs)
+        values['shipping_address_update_route'] = self._express_checkout_shipping_route
+        return values
+
+    def _get_shop_payment_values(self, order, **kwargs):
```

Comment:
This method has to be fixed in `main.py`, as most of the logic is already there. No need for an override.
From what I read, only the loop on the `delivery_carriers` is necessary